File: lib/models/custom_list.dart
class CustomList {
  final String id;
  final String name;
  final List<String> itemImdbIds;

  CustomList({
    required this.id,
    required this.name,
    required this.itemImdbIds,
  });

  factory CustomList.fromJson(Map<String, dynamic> json) {
    return CustomList(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      itemImdbIds: List<String>.from(json['items'] ?? []),
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'items': itemImdbIds,
      };
}

File: backend/models/CustomList.js
const mongoose = require('mongoose');

const CustomListSchema = new mongoose.Schema(
  {
    userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
    name: { type: String, required: true, trim: true },
    items: [{ type: String }], // Array of imdbIds
  },
  { timestamps: true }
);

module.exports = mongoose.model('CustomList', CustomListSchema);